lib/ipcs.c

```diff
@@ -75,6 +77,18 @@ qb_ipcs_create(const char *name,
 	qb_list_init(&s->list);
 	qb_list_add(&s->list, &qb_ipc_services);
 
+	/* Randomise socket names */
+	fd = open("/dev/urandom", O_RDONLY);
+	if (fd == -1) {
+		seed = (time_t)time(NULL);
+	} else {
+		if (read(fd, &seed, sizeof(seed)) != 4) {
```

Comment:
This effectively gives 32 bits of randomness to the generator. Theoretically it's not impossible to break by brute forcing this kind of number space, though in practice it might be impossible for the attacker to generate 2^32 files in /dev/shm. I'd still try to find out a way to get at least 64 bits of randomness to the file names to be sure.

Reply:
srand() only takes an int! Should I not be using random - even with a seed from /dev/urandom?

Reply:
Here's a good implementation in [systemd src/basic/random-util.c](https://github.com/systemd/systemd/blob/ca66b69d5bdc9d3bf34cdc53bb600f88ef6a2c87/src/basic/random-util.c#L68), though it may need some porting to run on non-Linux. It's also a bit complex.